sql/dd/impl/bootstrap/bootstrap_ctx.h

```diff
@@ -100,6 +100,7 @@ class DD_bootstrap_ctx {
 
   uint m_did_I_S_upgrade_from = 0;
   uint m_actual_I_S_version = 0;
+  uint m_actual_dd_engine = 0;
```

Comment:
This needs a comment explaining what is the "actual DDSE". If I understand correctly, it is the one that at the moment happens to contain the DD tables, as indicated by the presence of `mysql.dd_properties`. It is assumed (again, if I understand correctly), that moving any single table between SEs is atomic, and so `dd_properties` will be present in exactly one SE at all times. The actual DDSE may be different from `default_dd_storage_engine` setting. There may be more subtlety still.